vpd-manager/src/manager.cpp

```diff
@@ -640,4 +645,42 @@ void Manager::performVpdRecollection()
         m_worker->performVpdRecollection();
     }
 }
+
+int Manager::performVpdSanityCheck(const types::Path i_vpdPath)
+{
+    try
+    {
+        if (i_vpdPath.empty())
+        {
+            throw std::runtime_error("Given FRU path is empty");
+        }
+
+        nlohmann::json l_sysCfgJsonObj{};
+
+        if (m_worker.get() != nullptr)
+        {
+            l_sysCfgJsonObj = m_worker->getSysCfgJsonObj();
+        }
+
+        std::shared_ptr<Parser> l_parserObj =
+            std::make_shared<Parser>(i_vpdPath, l_sysCfgJsonObj);
+
+        std::shared_ptr<vpd::ParserInterface> l_vpdParserInstance =
+            l_parserObj->getVpdParserInstance();
```

Comment:
I also could not see the flow where path is validated against JSON and redundant path is looked for.